tests/transforms/test_qcut.py

```diff
@@ -1959,6 +1962,79 @@ def test_mc_state_prep(self):
                 assert op.wires == exp_op.wires
 
 
+class TestMCPostprocessing:
+    """
+    Tests that the postprocessing for circuits containing sample measurements
+    gives the correct results.
+    """
+
+    def test_sample_postprocess(self):
```

Comment:
Might be good to have an interface-based test like [here](https://github.com/PennyLaneAI/pennylane/blob/482cd0894548adeb1d507d7debf84aa321cc0726/tests/transforms/test_qcut.py#L2212), where we also check that the returned is converted to the right interface type.